xml_analyzer/xmlreader.py

```diff
@@ -1,144 +1,153 @@
+        self._tree = Et.parse(self._project_file)
+        self._root = self._tree.getroot()
 
     @staticmethod
-    def parse_constraint_spec(constraint_spec: str) -> list:
+    def _parse_constraint_spec(constraint_spec: str) -> List:
         autocalc_sysml_type_name = list()
-        for method in XMLReader.autocalc_methods:
-            if method in constraint_spec:
-                autocalc_sysml_type_name.append(method)
-                for occurrence in constraint_spec.split(method)[1:]:
+        for method in AutocalcMethods:
+            if method.name in constraint_spec:
+                autocalc_sysml_type_name.append(method.name)
+                for occurrence in constraint_spec.split(method.name)[1:]:
                     search_param = occurrence[occurrence.find('(') + 1:occurrence.find(')')]
-                    if search_param != '':
+                    if search_param:
                         autocalc_sysml_type_name.append(search_param)
         return autocalc_sysml_type_name
 
-    def find_attributes(self, val: list, method='by_type'):
+    def _find_attributes(self, val: List, method='by_type'):
         elements = None
         if method == 'by_type':
             query = ".//Attribute/Type//*[@Name='{}']/../..".format(val)
-            elements = self.root.findall(query)
+            elements = self._root.findall(query)
 
         return elements
 
-    def resolve_bindings(self, constraint_property: Et.Element) -> list:
+    def _resolve_bindings(self, constraint_property: Et.Element) -> List:
         query = ".//SysMLConstraintProperty[@Id='{}']//SysMLConstraintBlock".format(constraint_property.get("Id"))
-        constraint_block = self.root.find(query)
+        constraint_block = self._root.find(query)
         query = ".//SysMLConstraintBlock[@Id='{}']//Attribute//SysMLBindingConnector"\
             .format(constraint_block.get("Idref"))
-        binding_connectors_refs = self.root.findall(query)
+        binding_connectors_refs = self._root.findall(query)
         binding_connectors = list()
         for ref in binding_connectors_refs:
             ref_id = ref.get('Idref')
-            con = self.root.find(".//SysMLBindingConnector[@Id='{}']".format(ref_id))
+            con = self._root.find(".//SysMLBindingConnector[@Id='{}']".format(ref_id))
             binding_connectors.append(con)
 
         return binding_connectors
 
-    def resolve_dependencies(self, binding_connector_list: list) -> list:
+    def _resolve_dependencies(self, binding_connectors: List) -> List:
         dependency = collections.namedtuple('dependency', ['property', 'constraint_property_id'])
         dependencies = list()
-        for binding_connector in binding_connector_list:
+        for binding_connector in binding_connectors:
             stereo = binding_connector.find('./Stereotypes/Stereotype[@Name="external"]')
             if stereo is not None:
```

Comment:
can be simplified to if stereo

Reply:
Fixed with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/ffeb519bb77b6dd72bd2e24167623a3a70184ea1

Reply:
Turn out this does not work. Reverted to former state with https://github.com/Squiddrone/VPParametricsSolver/pull/2/commits/5e9632342512a874b9746546b01a3087305c6dd9